src/main/java/org/humancellatlas/ingest/project/ProjectChangeListener.java

```diff
@@ -0,0 +1,40 @@
+package org.humancellatlas.ingest.project;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.humancellatlas.ingest.core.exception.MultipleOpenSubmissions;
+import org.humancellatlas.ingest.messaging.MessageRouter;
+import org.humancellatlas.ingest.submission.SubmissionEnvelope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.mapping.event.AbstractMongoEventListener;
+import org.springframework.data.mongodb.core.mapping.event.BeforeSaveEvent;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Component
+@RequiredArgsConstructor
+@Getter
+public class ProjectChangeListener extends AbstractMongoEventListener<Project> {
+    @Autowired
+    @NonNull
+    private final MessageRouter messageRouter;
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    protected Logger getLog() {
+        return log;
+    }
+
+    @Override
+    public void onBeforeSave(BeforeSaveEvent<Project> event) {
```

Comment:
forget about my onBeforeCreate comment. the event doesn't seem to be in the latest spring-data mongodb lib anymore. Alternatively we could use ValidatingRepositoryEventListener for this type of check.

Reply:
I originally used ValidatingRepositoryEventListener but from my testing, it doesn't seem to work for dbref links updates.